packages/grafana-ui/src/types/datasource.ts

```diff
@@ -437,6 +437,14 @@ export interface ScopedVars {
   [key: string]: ScopedVar;
 }
 
+export interface StreamingQueryOptions {
+  /**
+   * Request an explicit buffer size.  When not specified, this will
+   * default to `request.maxDataPoints`
+   */
+  buffer?: number;
```

Comment:
I wonder if this makes sense as a separate option, could we just use `maxDataPoints` and say, for streaming/live targets it work as a circular buffer? Otherwise it would not be clear what happens when you set both.